indexer/src/main.ts

```diff
@@ -82,14 +103,14 @@ export default async function transform({
 
   const store: Array<StoreItem> = [];
 
-  await Promise.all(
-    (events ?? []).map(async ({ transaction, receipt, event }) => {
-      // Can be false if the transaction is not related to a specific instance of the Kakarot contract.
-      // This is typically the case if there are multiple Kakarot contracts on the same chain.
+  const maybeTrieData: Array<TrieData | null> = (events ?? []).map(
```

Comment:
not really in the scope of this PR but I don't really like this function
1. I don't like object mutation (like `store`) in a `map`
1. Couldn't we use some initial `.filter` instead of early returns in the `.map`?

Is there something wrong with (not tried)

```
events
    .filter((event) => !!isKakarotTransaction(transaction))
    .filter((event) => !ethValidationFailed(event.event))
    .map((event) => ({...event, transaction: toTypedEthTx(event.transaction)}))
    .filter((event) => !!event.transaction)
    .map((event) => ({...event, transaction: typedTransactionToEthTx({
        typedTransaction: typedEthTx,
        event.receipt,
        blockNumber,
        blockHash,
        isPendingBlock,
      })}))
    .filter((event) => !!event.transaction)
    .map((event) => ({...event, ethLogs: event.receipt.events
        .map((e) => {
          return toEthLog({
            transaction: ethTx,
            event: e,
            blockNumber,
            blockHash,
            isPendingBlock,
          });
        })
        .filter((e) => e !== null)
        .map((log: JsonRpcLog, index) => {
        log.logIndex = index.toString();
        return log;
      })}))
    .map((event) => ({...event, receipt: toEthReceipt({
        transaction: ethTx,
        logs: ethLogs,
        event,
        cumulativeGasUsed,
        blockNumber,
        blockHash,
        isPendingBlock,
      })}))
```

etc.

Reply:
Yes you are right, I've opened an issue here to track this: https://github.com/kkrt-labs/kakarot-rpc/issues/1187

In my opinion, this should be part of a general cleanup of our indexer. Will open other follow up issues on this.

I would say that it is better to avoid solving this problem in this PR to avoid getting confused knowing that the changes here were already rather tricky to make.